binary-search-tree/node-js/inventory-management.js

```diff
@@ -18,16 +18,37 @@ class BinarySearchTree {
         this.root = null;
     }
 
-    search(id) {
-        console.error("you must implement search");
+    search(id, currentNode) {
+        if (id === null) {
+            return null;
+        } else {
+            return this.searchRecursive(id, currentNode)
+        }
     }
-
+        // console.error("you must implement search");
+        // 1. Start from the root node
+        // 2. Compare the current node with the search value. Return early if found
+        // 3. Determine if we should go left or right
+        // 4. Update the current node based on the decision from (3). Rinse and repeat from (2)
+    searchRecursive(id, currentNode) {
+        if(currentNode.val.id === id ) {
+            return currentNode
+        }
+        else if (id > currentNode.val.id) {
+            return this.searchRecursive(id, currentNode.right)
+        }
+        else (id < currentNode.val.id) 
```

Comment:
You've inlined the body of the last `else`. It's generally a good idea to give each one of the conditional branches a body explicitly defined by curly braces.  

While both forms are valid JavaScript syntax, using the curly braces to explicitly define the body of a conditional branch's statement can help to make the code more readable and reduce the potential for errors caused by ambiguity in the code.